x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigInitialNode.java

```diff
@@ -0,0 +1,335 @@
+        KeyPair httpKeyPair = CertGenUtils.generateKeyPair(HTTP_KEY_SIZE);
+        // non-CA
+        X509Certificate httpCert = CertGenUtils.generateSignedCertificate(certificatePrincipal,
+                subjectAltNames, httpKeyPair, httpCACert, httpCAKeyPair.getPrivate(), false, HTTP_CERTIFICATE_DAYS, null);
+
+        try (SecureString nodeKeystorePassword = new SecureString(terminal.readSecret("", KeyStoreWrapper.MAX_PASSPHRASE_LENGTH));
+             KeyStoreWrapper nodeKeystore = KeyStoreWrapper.bootstrap(env.configFile(), () -> nodeKeystorePassword)) {
+            Path transportKeystoreOutput = instantAutoConfigDir.resolve(TRANSPORT_AUTOGENERATED_KEYSTORE_NAME + ".p12");
+            try (SecureString transportKeystorePassword = keystorePassword()) {
+                KeyStore transportKeystore = KeyStore.getInstance("PKCS12");
+                transportKeystore.load(null);
+                // the PKCS12 keystore and the contained private key use the same password
+                transportKeystore.setKeyEntry(TRANSPORT_AUTOGENERATED_KEYSTORE_NAME, transportKeyPair.getPrivate(),
+                        transportKeystorePassword.getChars(), new Certificate[]{transportCert});
+                fullyWriteFile(transportKeystoreOutput, stream -> transportKeystore.store(stream, transportKeystorePassword.getChars()));
+                nodeKeystore.setString("xpack.security.transport.ssl.keystore.secure_password", transportKeystorePassword.getChars());
+            } finally {
+                nodeKeystore.save(env.configFile(), nodeKeystorePassword.getChars());
+            }
+            Path httpKeystoreOutput = instantAutoConfigDir.resolve(HTTP_AUTOGENERATED_KEYSTORE_NAME + ".p12");
+            try (SecureString httpKeystorePassword = keystorePassword()) {
+                KeyStore httpKeystore = KeyStore.getInstance("PKCS12");
+                httpKeystore.load(null);
+                // the keystore contains both the node's and the CA's private keys
+                // both keys are encrypted using the same password as the PKCS12 keystore they're contained in
+                httpKeystore.setKeyEntry(HTTP_AUTOGENERATED_KEYSTORE_NAME + "_ca", httpCAKeyPair.getPrivate(),
+                        httpKeystorePassword.getChars(), new Certificate[]{httpCACert});
+                httpKeystore.setKeyEntry(HTTP_AUTOGENERATED_KEYSTORE_NAME, httpKeyPair.getPrivate(),
+                        httpKeystorePassword.getChars(), new Certificate[]{httpCert, httpCACert});
+                fullyWriteFile(httpKeystoreOutput, stream -> httpKeystore.store(stream, httpKeystorePassword.getChars()));
+                nodeKeystore.setString("xpack.security.http.ssl.keystore.secure_password", httpKeystorePassword.getChars());
+            } finally {
+                nodeKeystore.save(env.configFile(), nodeKeystorePassword.getChars());
+            }
+        }
+
+        {
```

Comment:
The TLS keystores are practically unusable outside the node because they are password protected.
This is a problem if you simply need a CA cert (no key) to configure a client. We discussed cmd line clients that don't have to prompt for the node keystore password.

Reply:
> We discussed cmd line clients that don't have to prompt for the node keystore password.

Those would use the HTTP truststore, right ? 

Fine to keep this anyhow, my question was mostly for the sake of discussion

Reply:
The lines shifted and it wasn't clear to me which truststore you referred to.

For the transport one, I only recall that I disliked having to set a password for a truststore. On the other hand, this creates two files that must always be used together, and the whole purpose of using PKCS12 is to have fewer of these files. I think having a single file for transport TLS actually conveys more that internode relays on a "shared secret" between the nodes, rather than public key infrastructure. 

I'll change it to use a single file.